File: adapter/java/AndroidPhoneAdapter.java
package androidhot.github.io.designpatterndemo.adapter.java;

public class AndroidPhoneAdapter implements ApplePhone {
    private AndroidPhone androidPhone;

    public AndroidPhoneAdapter() {
        androidPhone = new AndroidPhone();
    }

    @Override public void operatingSystem() {
        androidPhone.androidSystem();
    }
}

File: adapter/kotlin/AndroidPhoneAdapter.kt
package androidhot.github.io.designpatterndemo.adapter.kotlin

public class AndroidPhoneAdapter: ApplePhone {
    private var androidPhone: AndroidPhone? = null

    init {
        this.androidPhone = AndroidPhone()
    }

    override fun operatingSystem() {
        androidPhone?.androidSystem()
    }
}
